### PR TITLE
DOC: Fix docstring for read_sql_table

### PR DESCRIPTION
**Pull request by igblackadder**
- [ ] closes #xxxx
- [ ] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

```
################################################################################
###################### Docstring (pandas.read_sql_table)  ######################
################################################################################

Read SQL database table into a DataFrame.

Given a table name and a SQLAlchemy connectable, returns a DataFrame.
This function does not support DBAPI connections.

Parameters
----------
table_name : str
    Name of SQL table in database.
con : SQLAlchemy connectable or database URI str
    SQLite DBAPI connection mode not supported.
schema : str, default None
    Name of SQL schema in database to query (if database flavor
    supports this). Uses default schema if None (default).
index_col : str or list of str, optional, default: None
    Column(s) to set as index(MultiIndex).
coerce_float : bool, default True
    Attempts to convert values of non-string, non-numeric objects (like
    decimal.Decimal) to floating point. Can result in loss of Precision.
parse_dates : list or dict, default None
    The behavior is as follows:
    - List of column names to parse as dates.
    - Dict of ``{column_name: format string}`` where format string is
      strftime compatible in case of parsing string times or is one of
      (D, s, ns, ms, us) in case of parsing integer timestamps.
    - Dict of ``{column_name: arg dict}``, where the arg dict corresponds
      to the keyword arguments of :func:`pandas.to_datetime`
      Especially useful with databases without native Datetime support,
      such as SQLite.
columns : list, default None
    List of column names to select from SQL table.
chunksize : int, default None
    If specified, returns an iterator where `chunksize` is the number of
    rows to include in each chunk.

Returns
-------
DataFrame
    A SQL table is returned as two-dimensional data structure with labeled
    axes.

See Also
--------
read_sql_query : Read SQL query into a DataFrame.
read_sql : Read SQL query or database table into a DataFrame.

Notes
-----
Any datetime values with time zone information will be converted to UTC.

Examples
--------
>>> pd.read_sql_table('table_name', 'postgres:///db_name')  # doctest:+SKIP

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.read_sql_table" correct. :)
```

### CODE REVIEW

**Comment by gfyoung** [comment]
@igblackadder : This looks great!  Can you merge / rebase your branch with the most recent `master` ?  That will the test failures.

@datapythonista : Don't we have to add this file for its docstrings to get linted, or are there other docstring issues that still need to be addressed in `pandas/io/sql.py` ?

**Comment by WillAyd** [comment]
@gfyoung might be something I'm overlooking but I think the validation runs for all docstrings but a subset of codes

https://github.com/pandas-dev/pandas/blob/c9863865c217867583e8f6592ba88d9200601992/ci/code_checks.sh#L245

**Review by WillAyd — APPROVED** [review]
lgtm - @gfyoung I'd say go ahead and merge when CI is green

File: pandas/io/sql.py [hunk 1]
```diff
@@ -182,26 +182,28 @@ def execute(sql, con, cur=None, params=None):
...
     Given a table name and a SQLAlchemy connectable, returns a DataFrame.
     This function does not support DBAPI connections.
 
     Parameters
     ----------
-    table_name : string
+    table_name : str
         Name of SQL table in database.
-    con : SQLAlchemy connectable (or database string URI)
+    con : SQLAlchemy connectable or database URI str
```
Comment (datapythonista) [inline pandas/io/sql.py]: in this line we ideally would like to have just types (like type annotations). Not sure what the SQLAlchemy type would be, but for the database URI str, just leave `str`, and explain that is the database URI in the description of the parameter

**Review by datapythonista — COMMENTED** [review]
looks good, just one minor thing, and as it was said, you'll have to update your branch (i.e. `git fetch upstream && git merge upstream/master`)

**Comment by datapythonista** [comment]
@gfyoung, what @WillAyd said is correct. For now we're still focussing on getting as many formatting errors fixed for all docstrings. After we've got everything that is pure formatting (capitalization, missing full stop...), we'll probably switch to have some groups of docstrings (e.g. `pandas.Series.*`) fully validated.

**Comment by gfyoung** [comment]
Looks like all comments have been addressed, so merging.

**Comment by gfyoung** [comment]
Thanks @igblackadder !

**Comment by igblackadder** [comment]
Thank you @gfyoung and @WillAyd for your helpful comments. @datapythonista thank you so much for time and insight on Wednesday and for your comments on this PR!

**Pull request merged**